### PR TITLE
Add PointerOf Modifier

### LINKED ISSUE #56: A modifier for pointer values
Manually managed primitives are unsafe; because we can't rely on user deallocating anything, make a Modifier type that acts as a smart pointer to do it for them.

### PR DESCRIPTION
**Pull request by CodaFi**
Fixes #56 and a few retain cycles in arrow gen.

### CODE REVIEW

File: SwiftCheck/Modifiers.swift [hunk 1]
```diff
@@ -239,15 +276,15 @@ public struct ArrowOf<T : protocol<Hashable, CoArbitrary>, U : Arbitrary> : Arbi
...
 	}
 
-	public init(_ arr : (T -> U)) {
+	public convenience init(_ arr : (T -> U)) {
 		self.init(Dictionary(), { (_ : T) -> U in return undefined() })
 
-		self.arr = { x in
-			if let v = self.table[x] {
+		self.arr = { [weak self] x in
+			if let v = self!.table[x] {
```
Comment (mpurland) [inline SwiftCheck/Modifiers.swift]: I think it would be safer to use self? rather than self!.

Comment (CodaFi) [inline SwiftCheck/Modifiers.swift]: Self is guaranteed to be alive in that call.  The ? is not any safer, so I should probably be using unretained or something.

Comment (mpurland) [inline SwiftCheck/Modifiers.swift]: perhaps [unowned self]

**Comment by CodaFi** [comment]
We're still leaking some 200 bytes in our test suite, but I think it's all coming from XCTest and its use of NSInvocation rather than us explicitly.

**Comment by CodaFi** [comment]
:boat:

**Pull request merged**